docs/contributing/index.rst

```diff
@@ -4,104 +4,325 @@
+To continue development, create a new development branch:
+
+    $ git checkout -b feature-branch
+
+7. Verify tat you are on the new branch:
+
+    $ git branch
+    * feature-branch
+    master
+
+8. Verify remote configurattion:
+
+    $ git remote -v
+    upstream	https://github.com/geomstats/geomstats (fetch)
+    upstream	https://github.com/geomstats/geomstats (push)
+    origin	    https://github.com/<username>/geomstats (fetch)
+    origin	    https://github.com/<username>/geomstats (push)
+
+
+Dependencies and a virtual environment
+---------------------------------------
+
+We recommend using the conda virtual environment to create separation between your development environment
+from aany other Geomstats versions installed on your system. Create a virtual environment with conda using:
+
+$ conda create -n geomstats-dev
+
+Then clone your code to this environment and activate it:
+
+    $ git clone git@github.com:<username>/geomstats.git
+    $ cd geomstats
+    $ conda activate geomstats-dev
+
+Now install the Geomstats minimumcrequirements:
+
+    $ conda env create --file environment.yml
+
+.. note::
+   See the `setup.cfg` file for details on all project requirements
+
+.. _backends dev
+
+Backends
+--------
+
+Geomstats supports several backends, namely:
+
+- numpy
+- autograd
+- tensorflow
+- pytorch
+
+The default backend is `numpy, therefore install the other backends using:
+
+$ pip install geomstats[<backend_name>]
+
+And use an environment variable to set backed:
+
+$ export GEOMSTATS_BACKEND=<backend_name>
+
+Run the tests
+--------------
+
+Geomstats tests can be run in two ways, either using `nose` or one of the backends
+mentioned earlier.
+
+Using nose
+~~~~~~~~~~
+
+To run tests with Nose, first install the required packages:
+
+    $ pip install geomstats[dev]
+
+Then run all tests using:
+
+    $ nose2 tests
+
+You can run tests by file name:
+
+    $ nose2 tests.test__my_manifold
+
+Alternatively, run a particular test in a given test file:
+
+    $ nose2 tests.test__my_manifold.TestMyManifold.test_dimension
+
+Using a backend
+~~~~~~~~~~~~~~~~~
+
+We recommend that you run  all tests using the different backends discussed in 
+the `backends section <backends dev`_ . 
+
+Threfore install the backends as described before running all the tests using:
+
+    $ numpy tests
+
+Run a test file:
+
+    $ numpy tests.test__my_manifold
+
+Run a single tests method using:
+
+    $ numpy tests.test__my_manifold.TestMyManifold.test_dimension
+
+Build the doc
+--------------
+
+Documentation in the Geomstats project is implemented using `sphinx`.
+Install the sphinx dependecies using:
+
+$ pip install nbsphinx-link nbsphinx jupyter sphinx sphinx-gallery pydata-sphinx-theme
+
+Then while in the project root folder, build the docs using:
+
+$ cd docs
+$ make html
+
+
+Folder Structure
+=================
+
+As a contributor, there are a couple of top-level directories to take note of. 
```

Comment:
I don't think this sentence (line 183) is grammatically correct.  
What about "When you open the Geomstats github page, you will see the top-level directories of the package. Below is a description of each directory"

Reply:
Done